### PR TITLE
Adds view profile feature

### PR DESCRIPTION
**Pull request by allen-tran**
you already know

2022-08-06 at 7 14 25 PM

### CODE REVIEW

File: disc/cog.py [hunk 1]
```diff
@@ -14,6 +19,58 @@ class Descriptions:
...
+        self.api = Linkedin(LINKEDIN_ACCOUNT, LINKEDIN_PASSWORD)
+
+    @cog_ext.cog_slash(name="view-profile", description=Descriptions.POST_JOB)
+    async def _view_profile(self, ctx: SlashContext, profile):
+
+        await ctx.send("Hang on, grabbing details for you...", hidden=True)
+
+        get_profile = self.api.get_profile(profile)
+
+        time_string = ""
```
Comment (truongluan303) [inline disc/cog.py:42]: Should this make more sense as `exp_string` instead?

File: disc/cog.py [hunk 2]
```diff
@@ -14,6 +19,58 @@ class Descriptions:
...
+        final_color = discord.Colour(random_color)
+
+        embed_content = discord.Embed(
+            title=f"A look at {get_profile['firstName']} {get_profile['lastName']}'s profile.",
+            type="rich",
+            description=description,
+            color=final_color
+        )
+
+        embed_content.set_thumbnail(url=get_profile['displayPictureUrl'] + get_profile['img_100_100'])
```
Comment (truongluan303) [inline disc/cog.py]: Did we make sure 100x100 is always an option? Should we consider dynamically grabbing the image size instead?

**Review by truongluan303 — APPROVED** [review]
I think we can add this in first, and create a new PR to refactor this profile scraper to live in its own pack. Considering how we did not do input validation and edge cases checking, I think if we keep it in this file that will be very huge.

**Comment by truongluan303** [comment]
```
2022-08-07T00:47:04.085794+00:00 app[worker.1]: setup(self)
2022-08-07T00:47:04.085809+00:00 app[worker.1]: File "/app/disc/cog.py", line 146, in setup
2022-08-07T00:47:04.085974+00:00 app[worker.1]: bot.add_cog(Slash(bot))
2022-08-07T00:47:04.085977+00:00 app[worker.1]: File "/app/disc/cog.py", line 31, in __init__
2022-08-07T00:47:04.086080+00:00 app[worker.1]: self.linkedin_profile_api = LinkedInProfileAPI(
2022-08-07T00:47:04.086090+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.10/site-packages/linkedin_api/linkedin.py", line 37, in __init__
2022-08-07T00:47:04.086175+00:00 app[worker.1]: self.client.authenticate(username, password)
2022-08-07T00:47:04.086177+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.10/site-packages/linkedin_api/client.py", line 116, in authenticate
2022-08-07T00:47:04.086266+00:00 app[worker.1]: raise ChallengeException(data["login_result"])
2022-08-07T00:47:04.086301+00:00 app[worker.1]: linkedin_api.client.ChallengeException: CHALLENGE
2022-08-07T00:47:04.086311+00:00 app[worker.1]: 
2022-08-07T00:47:04.086318+00:00 app[worker.1]: The above exception was the direct cause of the following exception:
2022-08-07T00:47:04.086318+00:00 app[worker.1]: 
2022-08-07T00:47:04.086320+00:00 app[worker.1]: Traceback (most recent call last):
2022-08-07T00:47:04.086321+00:00 app[worker.1]: File "/app/app.py", line 13, in <module>
2022-08-07T00:47:04.086391+00:00 app[worker.1]: bot.load_extension("disc.cog")
2022-08-07T00:47:04.086405+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 678, in load_extension
2022-08-07T00:47:04.086629+00:00 app[worker.1]: self._load_from_module_spec(spec, name)
2022-08-07T00:47:04.086631+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.10/site-packages/discord/ext/commands/bot.py", line 623, in _load_from_module_spec
2022-08-07T00:47:04.086847+00:00 app[worker.1]: raise errors.ExtensionFailed(key, e) from e
2022-08-07T00:47:04.086878+00:00 app[worker.1]: discord.ext.commands.errors.ExtensionFailed: Extension 'disc.cog' raised an error: ChallengeException: CHALLENGE
2022-08-07T00:47:04.258127+00:00 heroku[worker.1]: Process exited with status 1
```
Reverted this commit due to this error in prod

**Comment by truongluan303** [comment]
https://github.com/tomquirk/linkedin-api/issues/162
This prob seems to be rooted from the library we are using.
In the worst case, we probably have to implement our own scraping tool.

**Comment by truongluan303** [comment]
@allen-tran I will implement our own scraping instead. Looks like linkedin_api is actually not that reliable

**Comment by allen-tran** [comment]
> @allen-tran I will implement our own scraping instead. Looks like linkedin_api is actually not that reliable

cool i agree lets work on implementing it

**Pull request merged**